pyneuroml/neuron/__init__.py

```diff
@@ -126,3 +135,558 @@ def export_to_neuroml1(hoc_file, nml1_file_name, level=1, validate=True):
+        'mechanisms': {
+        },
+    }
+
+    if numEk > 0:
+        infodict['k_ion'] = {
+            'nsecs': numEk,
+            'avg_rev_pot': (totEk / numEk),
+            'int_conc': (totki / numEk),
+            'ext_conc': (totko / numEk),
+        }
+    else:
+        infodict['k_ion'] = {
+            'nsecs': numEk,
+            'avg_rev_pot': 'NA',
+            'int_conc': 'NA',
+            'ext_conc': 'NA',
+        }
+
+    if numEna > 0:
+        infodict['na_ion'] = {
+            'nsecs': numEna,
+            'avg_rev_pot': (totEna / numEna),
+            'int_conc': (totnai / numEna),
+            'ext_conc': (totnao / numEna),
+        }
+    else:
+        infodict['na_ion'] = {
+            'nsecs': numEna,
+            'avg_rev_pot': 'NA',
+            'int_conc': 'NA',
+            'ext_conc': 'NA',
+        }
+
+    if numEca > 0:
+        infodict['ca_ion'] = {
+            'nsecs': numEca,
+            'avg_rev_pot': (totEca / numEca),
+            'int_conc': (totcai / numEca),
+            'ext_conc': (totcao / numEca),
+        }
+    else:
+        infodict['ca_ion'] = {
+            'nsecs': numEca,
+            'avg_rev_pot': 'NA',
+            'int_conc': 'NA',
+            'ext_conc': 'NA',
+        }
+
+    # https://neuronsimulator.github.io/nrn/python/modelspec/programmatic/mechtype.html#MechanismType
+    mt = h.MechanismType(0)
+    mname = h.ref('')
+    pname = h.ref('')
+
+    for i in range(mt.count()):
+        mt.select(i)
+        mt.selected(mname)
+        # provide storage for mechanism
+        # https://neuronsimulator.github.io/nrn/python/programming/mechstan.html#MechanismStandard
+        ms = h.MechanismStandard(mname, 1)
+        numParams = ms.count()
+
+        # construct empty dicts to hold information
+        totParamVal = {}
+        paramsectiondict = {}  # type: dict[typing.Any, typing.Any]
+        for j in range(numParams):
+            # assign pname the name of the jth parameter
+            ms.name(pname, j)
+            totParamVal[j] = 0
+            paramsectiondict[pname[0]] = {}
+
+        numSecPresent = 0
+        for sec in seclist:
+            if (h.ismembrane(mname, sec=sec)):
+                numSecPresent += 1
+                # segment information is provided as a fraction of the total
+                # section length, not the segment list
+                seginfo = {}
+                for seg in range(1, sec.nseg + 1):
+                    mid_pt = get_seg_midpoint(seg, sec.nseg)
+                    logger.debug(f"section {sec}: {seg}/{sec.nseg}: {mid_pt}")
+                    ms._in(mid_pt, sec=sec)
+                    for j in range(numParams):
+                        ms.name(pname, j)
+                        totParamVal[j] += ms.get(pname)
+                        if not (str(sec)) in paramsectiondict[pname[0]].keys():
+                            paramsectiondict[pname[0]][str(sec)] = {}
+                        seginfo[seg] = ms.get(pname)
```

Comment:
```suggestion
                        seginfo[pname[0]] = ms.get(pname)
```
Because of this `seginfo` was getting overwritten for each of the parameters in a mechanism which emitting the last parameter as values for each of the parameters.

I still dont know why is this throwing the same `gmax` values for ih, nat channel with both being `nonhomogenous`. I will try running this function on neuroml generated neuron and see.

Reply:
I think having a dict with keys as `segment_id` and `pname[0]` instead of `seginfo` which just has keys as `segment_id` will be better.

Reply:
(should be) fixed in latest commits.